openghg/analyse/_scenario.py

```diff
@@ -0,0 +1,1453 @@
+                      species: Optional[str] = None,
+                      footprint: Optional[FootprintData] = None) -> None:
+        """
+        Add footprint data based on keywords or direct FootprintData object.
+        """
+        from openghg.util import clean_string
+
+        # Search for footprint data based on keywords
+        # - site, domain, inlet (can extract from obs), model, metmodel
+        if site and footprint is None:
+            site = clean_string(site)
+            if inlet is None and self.obs is not None:
+                inlet = self.obs.metadata["inlet"]
+
+            # TODO: Add case to deal with "multiple" inlets
+            if inlet == "multiple":
+                raise ValueError("Unable to deal with multiple inlets yet:\n Please change date range or specify a specific inlet")
+
+            footprint_keywords_1 = {"site": site,
+                                    "height": inlet,
+                                    "domain": domain,
+                                    "model": model,  # Not currently used in get_footprint - should be added
+                                    # "metmodel": metmodel,  # Should be added to inputs for get_footprint()
+                                    "start_date": start_date,
+                                    "end_date": end_date,
+                                    "species": species}
+
+            footprint_keywords_2 = footprint_keywords_1.copy()
+            footprint_keywords_2.pop("species")
+
+            footprint_keywords = [footprint_keywords_1, footprint_keywords_2]
+
+            footprint = self._get_data(footprint_keywords, input_type="footprint")
+
+        self.footprint = footprint
+
+        if self.footprint is not None and not hasattr(self, "site"):
+            self.site = self.footprint.metadata["site"]
+
+    def add_flux(self,
+                 species: Optional[str] = None,
+                 domain: Optional[str] = None,
+                 sources: Optional[Union[str, Sequence]] = None,
+                 start_date: Optional[Union[str, Timestamp]] = None,
+                 end_date: Optional[Union[str, Timestamp]] = None,
+                 flux: Optional[Union[FluxData, Dict[str, FluxData]]] = None) -> None:
+        """
+        Add flux data based on keywords or direct FluxData object.
+        Can add flux datasets for multiple sources.
+        """
+        if self.fluxes is not None:
+            # Check current species in any flux data
+            if species is not None:
+                current_flux_1 = list(self.fluxes.values())[0]
+                current_species = current_flux_1.metadata["species"]
+                if species != current_species:
+                    raise ValueError(f"New data must match current species {current_species} in ModelScenario. Input value: {species}")
+
+        if species and flux is None:
+            flux = {}
+
+            if sources is None or isinstance(sources, str):
```

Comment:
Because "sources" is being looped over and the each source used within `get_flux`. The code below needs a list / Iterable to be able to perform this loop. Passing source=None is fine for the underlying function.